python/gtsam/tests/test_Triangulation.py

```diff
@@ -15,66 +15,123 @@
-        pose1 = Pose3(upright, Point3(0, 0, 1))
-        camera1 = PinholeCameraCal3_S2(pose1, sharedCal)
+        self.pose1 = Pose3(upright, Point3(0, 0, 1))
 
         # create second camera 1 meter to the right of first camera
-        pose2 = pose1.compose(Pose3(Rot3(), Point3(1, 0, 0)))
-        camera2 = PinholeCameraCal3_S2(pose2, sharedCal)
+        self.pose2 = self.pose1.compose(Pose3(Rot3(), Point3(1, 0, 0)))
 
         # landmark ~5 meters infront of camera
-        landmark = Point3(5, 0.5, 1.2)
+        self.landmark = Point3(5, 0.5, 1.2)
+
+    
+    def test_TriangulationExample(self):
+        # Some common constants
+        sharedCal = Cal3_S2(1500, 1200, 0, 640, 480)        
+        camera1 = PinholeCameraCal3_S2(self.pose1, sharedCal)
+        camera2 = PinholeCameraCal3_S2(self.pose2, sharedCal)
 
         # 1. Project two landmarks into two cameras and triangulate
-        z1 = camera1.project(landmark)
-        z2 = camera2.project(landmark)
+        z1 = camera1.project(self.landmark)
+        z2 = camera2.project(self.landmark)
 
         # twoPoses
         poses = Pose3Vector()
         measurements = Point2Vector()
 
-        poses.append(pose1)
-        poses.append(pose2)
+        poses.append(self.pose1)
+        poses.append(self.pose2)
         measurements.append(z1)
         measurements.append(z2)
 
         optimize = True
         rank_tol = 1e-9
 
         triangulated_landmark = triangulatePoint3(poses,sharedCal, measurements, rank_tol, optimize)
-        self.gtsamAssertEquals(landmark, triangulated_landmark,1e-9)
+        self.gtsamAssertEquals(self.landmark, triangulated_landmark,1e-9)
 
         # 2. Add some noise and try again: result should be ~ (4.995, 0.499167, 1.19814)
         measurements = Point2Vector()
         measurements.append(z1 - np.array([0.1, 0.5]))
         measurements.append(z2 - np.array([-0.2, 0.3]))
 
         triangulated_landmark = triangulatePoint3(poses,sharedCal, measurements, rank_tol, optimize)
-        self.gtsamAssertEquals(landmark, triangulated_landmark,1e-2)
+        self.gtsamAssertEquals(self.landmark, triangulated_landmark,1e-2)
         #
         # # two Poses with Bundler Calibration
         # bundlerCal = Cal3Bundler(1500, 0, 0, 640, 480)
         # camera1 = PinholeCameraCal3Bundler(pose1, bundlerCal)
         # camera2 = PinholeCameraCal3Bundler(pose2, bundlerCal)
         #
-        # z1 = camera1.project(landmark)
-        # z2 = camera2.project(landmark)
+        # z1 = camera1.project(self.landmark)
+        # z2 = camera2.project(self.landmark)
         #
         # measurements = Point2Vector()
         # measurements.append(z1)
         # measurements.append(z2)
         #
         # triangulated_landmark = triangulatePoint3(poses,bundlerCal, measurements, rank_tol, optimize)
-        # self.gtsamAssertEquals(landmark, triangulated_landmark,1e-9)
+        # self.gtsamAssertEquals(self.landmark, triangulated_landmark,1e-9)
```

Comment:
I wonder why there is so much commented code here, can this be removed?

Reply:
that is an example of triangulation with bundler calibration. Have left it as it was. @ProfFan , should I put this into a different unit test?

Reply:
removed the commented code for now, since it is similar to Cal3_S2, and there is a unittest that uses distinct Bundler calibrations. Triangulation with a shared Bundler calibration can be rewitten as a separate unit test, if required